### PR TITLE
Fix test-in-docker

### PR DESCRIPTION
**Pull request by dritter**
This fixes the `test-in-docker` script and adds more recent ZSH versions (5.3.1 from Debian; rest is from Ubuntu).

//cc @docwhat

### CODE REVIEW

File: test-in-docker [hunk 1]
```diff
@@ -162,7 +162,7 @@ while (( $# > 0 )); do
       if [[ -z "$use_framework" ]]; then
         local f="$(resolve_framework "$1")"
         if [[ -n "$f" ]]; then
-          use_framework=$f
+          use_framework="${f}"
```
[the remark this answers is not in the document]
Comment (dritter) [reply test-in-docker]: Yes. That was done by force of habit.. 🙄

File: test-in-docker [hunk 2]
```diff
@@ -40,8 +40,8 @@ err()
 resolve_framework() {
   local f=$1 found
   found=${frameworks[(In:-1:)$f*]}
-  if (( found <= $#frameworks )); then
-    echo "${frameworks[$found]}"
+  if (( found <= ${#frameworks} )); then
+    echo "${1}"
```
[the remark this answers is not in the document]
Comment (dritter) [reply test-in-docker]: Hah! Good point!

File: test-in-docker [hunk 3]
```diff
@@ -40,8 +40,8 @@ err()
 resolve_framework() {
   local f=$1 found
   found=${frameworks[(In:-1:)$f*]}
-  if (( found <= $#frameworks )); then
-    echo "${frameworks[$found]}"
+  if (( found <= ${#frameworks} )); then
```
[the remark this answers is not in the document]
Comment (dritter) [reply test-in-docker]: Yes, force of habit as well. 😄

File: test-in-docker (hunk 2 above)
[the remark this answers is not in the document]
Comment (docwhat) [reply test-in-docker]: I fixed it in #884.  If you could review it, that'd be nice.

File: docker/prezto/install.zsh [hunk 4]
```diff
@@ -9,7 +9,7 @@ for rcfile in "${ZDOTDIR:-$HOME}"/.zprezto/runcoms/^README.md(.N); do
   ln -nsf "$rcfile" "${ZDOTDIR:-$HOME}/.${rcfile:t}"
 done
 
-ln -s "${HOME}/p9k/powerlevel9k.zsh-theme" \
+ln -sf "${HOME}/p9k/powerlevel9k.zsh-theme" \
```
[the remark this answers is not in the document]
Comment (dritter) [reply docker/prezto/install.zsh]: Good to know. 👍

File: test-in-docker (hunk 2 above)
[the remark this answers is not in the document]
Comment (dritter) [reply test-in-docker]: I'll do. And create a new PR with the new Dockerfiles on top of yours.

File: docker/prezto/install.zsh (hunk 4 above)
[the remark this answers is not in the document]
Comment (docwhat) [reply docker/prezto/install.zsh]: This is bigger deal with symlinks to directories.  Without the `-n` you build loops and other fun things.

**Comment by dritter** [comment]
Closed in favor of #884

**Pull request closed without merging**